### PR TITLE
Fix some typos, add 2 bonus questions.

### PR DESCRIPTION
**Pull request by salimfadhley**

### CODE REVIEW

File: question.0.shopping_basket.md [hunk 1]
```diff
@@ -2,26 +2,30 @@
...
 
 ### Behavior
 
-* A basket may contain products from a "Catalogue".
-* Products may be added to or removed from the basket. 
-* Given the contents of the basket, and the current "Offers", the basket will determine which offers are applicable, calculate the sub-total, discount and total. 
-* The design of the shopping basket should accomodate changes to the catalogue and offers.
+* A basket may contain products from a ***catalogue***. The catalogue represents all of the products currently sold by the supermarket. Every product in catalogue has a name and a price. 
+* A basket can contain zero or more products.
+* The discount and therefor the total price is determined by the contents of the basket, the undiscounted price of the goods and the applicable offers.
```
Comment (vincecartsvane) [inline question.0.shopping_basket.md]: `therefore`

File: question.0.shopping_basket.md [hunk 2]
```diff
@@ -2,26 +2,30 @@
...
 
 Other considerations:
-* We are only interested in the behavior of the shopping basket component. You do not need to build an API, web-interface or database.
-* We expect that your solution can be implemented in plain python without any 3rd party libries. If you feel you need to include a library in your solution please provide a mechanism for us to quickly install it.
-* Please provide some instructions which will help us run and test your code.
-* Your solution should return the sub-total, discount and total to 2 decimal places as calculated for the basket contents.
+* We are only interested in the behavior of the shopping basket component. You do not need to build an API, web-interface or database. Try to stay focused on implementing the shopping basket, and especially the process by which we determine the total price of items in the basket.
+* We expect that your solution should be implemented in plain python. You do not need to use any 3rd party libraries, however if you do - please include a mechanism for us to automatically install any project requirments.
```
Comment (vincecartsvane) [inline question.0.shopping_basket.md]: `requirements`

File: question.0.shopping_basket.md [hunk 3]
```diff
@@ -2,26 +2,30 @@
 
 ## Requirements
 
-You are implementing a simple "shopping basket" for a supermarket. This is an important component of a shopping system whose responsibility is to calculate the final price of the items *contained* in the basket.
+You are implementing a simple "shopping basket" for a supermarket. This is an important component of a shopping system. This component's responsibility is to calculate the final price of the items contained in the basket.
+
+For any given basket, your solution should return the sub-total, discount and total, all accurate to two decimal places.
+
+Bonus questions are optional. They are suggestions for how you might extend the system in ways that demonstrate your skill.
```
Comment (JohnELawson) [inline question.0.shopping_basket.md]: if you have time, we dont want them taking too long

File: question.0.shopping_basket.md [hunk 4]
```diff
@@ -2,26 +2,30 @@
 
 ## Requirements
 
-You are implementing a simple "shopping basket" for a supermarket. This is an important component of a shopping system whose responsibility is to calculate the final price of the items *contained* in the basket.
+You are implementing a simple "shopping basket" for a supermarket. This is an important component of a shopping system. This component's responsibility is to calculate the final price of the items contained in the basket.
```
Comment (JohnELawson) [inline question.0.shopping_basket.md]: calculate the correct final price? hint test it

File: question.0.shopping_basket.md (hunk 2 above)
Comment (XavierBrinonECS) [inline question.0.shopping_basket.md]: plain python / vanilla javascript

File: question.0.shopping_basket.md [hunk 5]
```diff
@@ -2,26 +2,30 @@
...
+* A basket can contain zero or more products.
+* The discount and therefor the total price is determined by the contents of the basket, the undiscounted price of the goods and the applicable offers.  
+* As with all supermarkets, the prices of goods and the offers applicable to those goods may change from day to day.
 
 Other considerations:
-* We are only interested in the behavior of the shopping basket component. You do not need to build an API, web-interface or database.
-* We expect that your solution can be implemented in plain python without any 3rd party libries. If you feel you need to include a library in your solution please provide a mechanism for us to quickly install it.
-* Please provide some instructions which will help us run and test your code.
-* Your solution should return the sub-total, discount and total to 2 decimal places as calculated for the basket contents.
+* We are only interested in the behavior of the shopping basket component. You do not need to build an API, web-interface or database. Try to stay focused on implementing the shopping basket, and especially the process by which we determine the total price of items in the basket.
```
Comment (XavierBrinonECS) [inline question.0.shopping_basket.md]: plain python / vanilla javascript

File: question.0.shopping_basket.md (hunk 3 above)
Comment (salimfadhley) [reply question.0.shopping_basket.md]: Yes, I added a note to that effect.

File: question.0.shopping_basket.md (hunk 4 above)
Comment (salimfadhley) [reply question.0.shopping_basket.md]: Agreed

File: question.0.shopping_basket.md (hunk 5 above)
Comment (salimfadhley) [reply question.0.shopping_basket.md]: Agreed

**Pull request merged**